bd-resilient-kv/src/versioned_kv_journal/store.rs

```diff
@@ -387,6 +392,107 @@ impl PersistentStore {
   }
 
   async fn rotate_journal(&mut self) -> anyhow::Result<Rotation> {
+    self.rotate_journal_with_hint(0).await
+  }
+
+  /// Calculate the compacted size estimate by summing encoded sizes of all live entries.
+  fn calculate_compacted_size(&self) -> usize {
+    self
+      .cached_map
+      .iter()
+      .map(|(_, key, tv)| {
+        super::framing::Frame::compute_encoded_size(key.as_str(), tv.timestamp, &tv.value)
+      })
+      .sum()
+  }
```

Comment:
Not sure how often this happens but if it's often we might want to just keep a running tally of this.

Reply:
Only on rotation or capacity reached, but would be super cheap to track so seems worth it? Will do as part of a follow up PR